### PR TITLE
Fix clippy for rust 1.40 and signal busy if dkg runing

### PR DESCRIPTION
**Pull request by jeanphilippeD**

### CODE REVIEW

File: src/parsec.rs [hunk 1]
```diff
@@ -446,10 +446,11 @@ impl<T: NetworkEvent, S: SecretId> Parsec<T, S> {
 
     /// Check if there are any observations that have been voted for but not yet polled - that is,
     /// either they haven't been consensused yet or a block containing that observation hasn't yet
-    /// been retrieved by calling `poll`.
+    /// been retrieved by calling `poll`, or a DKG is running.
     pub fn has_unpolled_observations(&self) -> bool {
         self.observations.values().any(|info| !info.consensused)
             || !self.consensused_blocks.is_empty()
+            || !self.key_gen.is_empty()
```
Comment (maqi) [inline src/parsec.rs]: just to confirm that routing integration tests passes using real parsec with this change ?

File: src/dump_graph.rs [hunk 2]
```diff
@@ -1193,7 +1193,12 @@ mod detail {
         fn new_peer_id_peer_index_map(values: &[(usize, &str)]) -> PeerIndexMap<String> {
             values
                 .iter()
-                .map(|(index, string)| (PeerIndex::new_test_peer_index(*index), string.to_string()))
+                .map(|(index, string)| {
+                    (
+                        PeerIndex::new_test_peer_index(*index),
+                        (*string).to_string(),
```
[the remark this answers is not in the document]
Comment (jeanphilippeD) [reply src/dump_graph.rs]:
```
Checking parsec v0.5.0 (/home/jp/Git/maidsafe/hgproj/parsec)
error: calling `to_string` on `&&str`
    --> src/dump_graph.rs:1196:81
     |
1196 |                 .map(|(index, string)| (PeerIndex::new_test_peer_index(*index), string.to_string()))
     |                                                                                 ^^^^^^^^^^^^^^^^^^ help: try dereferencing the receiver: `(*string).to_string()`
     |
note: lint level defined here
    --> src/lib.rs:122:5
     |
122  |     warnings
     |     ^^^^^^^^
     = note: `#[forbid(clippy::inefficient_to_string)]` implied by `#[forbid(warnings)]`
     = help: `&str` implements `ToString` through a slower blanket impl, but `str` has a fast specialization of `ToString`
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#inefficient_to_string

error: aborting due to previous error
```

File: src/parsec.rs (hunk 1 above)
Comment (jeanphilippeD) [reply src/parsec.rs]: yes, mentioned in commit comment.

File: src/dump_graph.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (maqi) [reply src/dump_graph.rs]: Ah, I see. thx

**Pull request merged**